chromoscores/snipping.py

```diff
@@ -0,0 +1,197 @@
+import numpy as np
+
+
+def peak_snipping(contact_map, window_size, peak_coordinate):
+    """
+    Peak snippet; snippet with a "window_size" around peak with "peak_coordinate"
+    on a contact map, "contact_map". peak_coordinate should be in the format
+     of (i,j)
+    -----------------
+    Function peak_snippet(contact_map, window_size, peak_coordinate):
+
+    begin function
+
+         raise an error if peak coordinate + window window_size is out of the map
+
+         snippet = contact_map[
+        (peak_coordinate[0] - window_size):(peak_coordinate[0] + window_size),
+        (peak_coordinate[1] - window_size):(peak_coordinate[1] + window_size)]
+
+     return snippet_matrix
+
+     end function
+    ----------------
+    """
+
+    if (peak_coordinate[1] + window_size) > len(contact_map) or (
+        peak_coordinate[0] - window_size
+    ) < 0:
+
+        raise ValueError(
+            "window window_size for peak coordinate exceeds window_size of the contact map"
+        )
+
+    snippet = contact_map[
+        (peak_coordinate[0] - window_size) : (peak_coordinate[0] + window_size),
+        (peak_coordinate[1] - window_size) : (peak_coordinate[1] + window_size),
+    ]
+
+    return snippet
+
+
+def tad_snipping(contact_map, stall_list, index):
+    """
+    Tad snippet
+    To extract Tad snippets adjacent to a boundary element with index of
+    "index" from a list of "stall_list":
+    -------------------------
+    Function tad_snippet(contact_map, stall_list, index):
+
+         begin function
+
+         raise an error if index is out of list range
+
+         set tad_matrix between sequential stalls starting with index:
+             tad = contact_map[
+                   stall_list[index]:stall_list[index + 1] + 1,
+                   stall_list[index]:stall_list[index + 1] + 1,
+                   ]
+
+    return tad_matrix
+
+    end function
+    """
+
+    if index + 1 > len(stall_list):
+        raise ValueError("index + 1 should be in the range of stall list")
+
+    tad = contact_map[
+        stall_list[index] : stall_list[index + 1] + 1,
+        stall_list[index] : stall_list[index + 1] + 1,
+    ]
+    return tad
+
+
+def tad_snippet_sectors(
```

Comment:
we should brainstorm a more informative function name.

Reply:
I changed it to tad_snippet_areas. areas here are area_in and area_out.